docs/guides/tutorials/cloudflare_workers.rst

```diff
@@ -0,0 +1,261 @@
+If you don't want to deploy up right away, just go to the project folder 
+you just made to start writing your code. When you're ready to put your 
+project on Cloudflare, you can run :code:`npx wrangler deploy` to push it.
+
+.. note:: Using Wrangler CLI
+
+    If you prefer using `Wrangler`_ to set up your worker, you can use the 
+    :code:`wrangler generate` command to create a new project.
+
+.. _Wrangler: https://developers.cloudflare.com/workers/cli-wrangler
+
+
+Configure EdgeDB
+=================
+
+You can use `EdgeDB Cloud`_ for a managed service or run EdgeDB locally.
+
+.. _`EdgeDB Cloud`: https://www.edgedb.com/cloud
+
+**Local EdgeDB Setup (Optional for EdgeDB Cloud Users)**
+
+If you're running EdgeDB locally, you can use the following command 
+to create a new instance:
+
+.. code-block:: bash
+
+    $ edgedb project init
+
+It creates an :code:`edgedb.toml` config file, and a schema file 
+:code:`dbschema/default.esdl`.
+
+It also spuns up an EdgeDB instance and “links” it to the current directory. 
+As long as you’re inside the project directory, all CLI commands will 
+be executed against this instance.
+
+You can run :code:`edgedb` in your terminal to open an 
+interactive REPL to your instance. 
+
+.. code-block:: bash
+
+    $ edgedb
+
+**Install The EdgeDB NPM package**
+
+.. code-block:: bash
+
+    $ npm install edgedb # or pnpm, yarn, bun
+
+**Extend The Default Schema (Optional)**
+
+You can extend the default schema, :code:`dbschema/default.esdl`, to define 
+your data model, and then try it out in the Cloudflare Worker code.
+
+Add new types to the schema file:
+
+.. code-block:: sdl
+
+    module default {
+      type Movie {
+        required property title -> str {
+          constraint exclusive;
+        };
+        multi link actors -> Person;
+      }
+
+      type Person {
+        required property name -> str;
+      }
+    }
+
+Then apply the schema schema to your EdgeDB instance:
+
+.. code-block:: bash
+
+    $ edgedb migration create
+    $ edgedb migrate
+
+Using EdgeDB in a Cloudflare Worker
+====================================
+
+Open the :code:`index.ts` file from the :code:`src` directory in your project, 
+and remove the default code.
+
+To interact with your **local EdgeDB instance**, use the following code:
+
+.. code-block:: typescript
+
+    import * as edgedb from 'edgedb';
+
+    export default {
+     async fetch(
+      _request: Request, 
+      env: Env, 
+      ctx: ExecutionContext
+     ): Promise<Response> {
+      
+      const client = edgedb.createHttpClient({
+       tlsSecurity: 'insecure',
+       dsn: '<your-edgedb-dsn>',
+      });
+
+      const movies = await client.query(`select Movie { title }`);
+
+      return new Response(JSON.stringify(movies, null, 2), {
+       headers: {
+        'content-type': 'application/json;charset=UTF-8',
+       },
+      });
+     },
+    } satisfies ExportedHandler<Env>;
```

Comment:
Can we add explicit imports for these various types?

Reply:
They are actually globals in the Cloudflare project. They add this config setting out-of-the-box when creating a new project:

```
"types": ["@cloudflare/workers-types/2023-07-01"]
```